packages/blade/src/components/Switch/_decisions/decisions.md

```diff
@@ -0,0 +1,69 @@
+# Switch API Decisions <!-- omit in toc -->
+
+A switch component is used to quickly switch between two possible states. These are only used for binary actions that occur immediately after the user turn the switch on/off.
+
+<img src="./switch-thumbnail.png" width="380" alt="" />
+
+- [Design](#design)
+- [`Switch` API](#switch-api)
+  - [Examples:](#examples)
+    - [Basic](#basic)
+    - [Controlled, Uncontrolled](#controlled-uncontrolled)
+- [Accessibility](#accessibility)
+- [Open Questions](#open-questions)
+
+## Design
+
+[Figma Link](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=31366-831710&t=qzTLJeIDluL4Ps23-0) to all variants of the Switch component
+
+
+## `Switch` API
+
+| Prop       | Type              | Default     | Description                 | Required |
+| ---------- | ----------------- | ----------- | --------------------------- | -------- |
+| isChecked | `boolean` | `false` | State of the switch, if explicitly set the switch will become [controlled component](https://reactjs.org/docs/forms.html#controlled-components) |        |
+| defaultChecked | `boolean` | `false` | Default state of the switch, if isChecked is not provided the switch will become [uncontrolled component](https://reactjs.org/docs/uncontrolled-components.html) |        |
+| onChange | `({ event, isChecked, value }) => void` | `undefined` | The function to be called when the switch state changes. |        |
+| isDisabled | `boolean` | `false` | Control whether the switch is disabled or not. |        |
+| accessibilityLabel | `string` | `undefined` | Provides accessible label for internal checkbox component that sets the aria-label prop for screen readers. |        |
+| size | `small, medium` | `medium` | Size of the switch component |        |
+| name (html) | `string` | `undefined` | The name of the input field in a switch, [useful in form submissions](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#name) |        |
+| value (html) | `string` | `undefined` | The value of the input field in a switch, [useful in form submissions](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#value) |        |
+
```

Comment:
Do we need label (see some usecases in design)

![image](https://github.com/razorpay/blade/assets/6682655/aedc55e9-e74d-4d3c-b94f-24f846756d60)

 validation states, helper texts etc.? eg if being used in form fields (though don't see these on design side)

Reply:
Yeah these are not in the design. I've asked from design side if they are needed.

Reply:
> Do we need label 

Discussing this internally.

Reply:
From design side we decided to not add labels, consumers can either use their own label or use accessibilityLabel when a visual label isn't appropriate or required.